src/careamics/dataset_ng/patch_extractor/image_stack/image_utils/zarr_utils.py

```diff
@@ -250,26 +230,48 @@ def create_zarr_image_stacks(
                 # collect all arrays
                 array_paths = collect_arrays(zarr_group)
 
+                # sort names
+                array_paths.sort()
+
             # instantiate image stacks
             for array_path in array_paths:
                 image_stacks.append(
                     ZarrImageStack(group=zarr_group, data_path=array_path, axes=axes)
                 )
 
-        elif is_file_uri(data_str):
+        elif is_zarr_uri(data_str):
             # decipher the uri and open the group
-            store_path, group_path, array_name = decipher_zarr_path(data_str)
+            store_path, parent_path, name = decipher_zarr_path(data_str)
+
+            zarr_group = zarr.open(store_path, mode="r")[parent_path]
```

Comment:
```suggestion
            zarr_group = zarr.open_group(store_path, path=parent_path, mode="r")
```
Pylance on strict is giving me type errors otherwise

Reply:
Also we don't have zarr as a mypy additional dependency in the .pre-commit-config.yaml so we're not checking the correctness of zarr types in the CI at the moment

Reply:
> Also we don't have zarr as a mypy additional dependency in the .pre-commit-config.yaml so we're not checking the correctness of zarr types in the CI at the moment

Ah good point! I added it.